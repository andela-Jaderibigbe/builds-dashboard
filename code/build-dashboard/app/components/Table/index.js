import React from 'react';
import PropTypes from 'prop-types';

import styles from './Table.scss';

const Table = ({ className, children }) => {
  className = `${styles.table} ${className}`;
  return <div className={className}>{children}</div>;
};

Table.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.array]),
};

export default Table;
