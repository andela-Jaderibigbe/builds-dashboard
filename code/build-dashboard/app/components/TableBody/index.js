import React from 'react';
import PropTypes from 'prop-types';
import styles from './TableBody.scss';

const TableBody = ({ className, children }) => {
  className = `${styles.table} ${className}`;
  return <div className={className}>{children}</div>;
};

TableBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default TableBody;
