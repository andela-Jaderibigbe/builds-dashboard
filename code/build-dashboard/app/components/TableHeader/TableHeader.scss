@import 'theme/variables';

.tableHeader {
  background-color: $verylightgray;
  border-bottom: 1px solid $gray;
}
