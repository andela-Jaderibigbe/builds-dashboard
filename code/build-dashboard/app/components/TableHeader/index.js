import React from 'react';
import PropTypes from 'prop-types';

import styles from './TableHeader.scss';

const TableHeader = ({ className, children }) => {
  className = `${styles.tableHeader} ${className}`;
  return <div className={className}>{children}</div>;
};

TableHeader.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.node]),
};

export default TableHeader;
