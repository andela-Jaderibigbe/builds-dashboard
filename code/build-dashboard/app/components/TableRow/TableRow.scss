.tableRow {
  width: 100%;
  display: flex;
  align-items: stretch;
  height: inherit;
  padding: 8px;
}
