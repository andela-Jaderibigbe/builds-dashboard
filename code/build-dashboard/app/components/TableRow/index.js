import React from 'react';
import PropTypes from 'prop-types';
import styles from './TableRow.scss';

const TableRow = ({ className, children }) => {
  className = `${styles.tableRow} ${className}`;
  return <div className={className}>{children}</div>;
};

TableRow.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default TableRow;
