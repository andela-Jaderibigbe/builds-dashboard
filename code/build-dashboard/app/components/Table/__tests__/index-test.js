import React from 'react';
import renderer from 'react-test-renderer';

import TableHeader from 'components/TableHeader';
import TableBody from 'components/TableBody';
import TableRow from 'components/TableRow';

import Table from '../';

const content = <div className="unique" />;

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Table>
        <TableHeader>{content}</TableHeader>
        <TableBody>
          <TableRow>{content}</TableRow>
        </TableBody>
      </Table>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
