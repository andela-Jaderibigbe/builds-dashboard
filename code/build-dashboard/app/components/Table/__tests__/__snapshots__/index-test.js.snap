// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="table undefined"
>
  <div
    className="tableHeader undefined"
  >
    <div
      className="unique"
    />
  </div>
  <div
    className="table undefined"
  >
    <div
      className="tableRow undefined"
    >
      <div
        className="unique"
      />
    </div>
  </div>
</div>
`;
